apps/design/backend/src/worker/tasks.ts

```diff
@@ -0,0 +1,68 @@
+import fs from 'fs/promises';
+import JsZip from 'jszip';
+import path from 'path';
+import { z } from 'zod';
+import { throwIllegalValue } from '@votingworks/basics';
+import { layOutAllBallotStyles } from '@votingworks/hmpb-layout';
+import {
+  BallotType,
+  ElectionPackageFileName,
+  getDisplayElectionHash,
+  Id,
+  safeParseJson,
+} from '@votingworks/types';
+
+import { PORT } from '../globals';
+import { BackgroundTask } from '../store';
+import { Workspace } from '../workspace';
+
+async function generateElectionPackage(
+  { assetDirectoryPath, store }: Workspace,
+  { electionId }: { electionId: Id }
+): Promise<void> {
+  const { election, layoutOptions, systemSettings } =
+    store.getElection(electionId);
+  const { electionDefinition } = layOutAllBallotStyles({
+    election,
+    // Ballot type and ballot mode shouldn't change the election definition, so it doesn't matter
+    // what we pass here
+    ballotType: BallotType.Precinct,
+    ballotMode: 'test',
+    layoutOptions,
+  }).unsafeUnwrap();
+  const zip = new JsZip();
+  zip.file(ElectionPackageFileName.ELECTION, electionDefinition.electionData);
+  zip.file(
+    ElectionPackageFileName.SYSTEM_SETTINGS,
+    JSON.stringify(systemSettings, null, 2)
+  );
```

Comment:
This logic above is unchanged, just moved from `app.ts`.

The logic below is what's different. Whereas the backend used to send the entire zip contents to the frontend in its response, we now store the zip in the backend's file system and return a URL to the frontend that the frontend can use to download it.

When we eventually deploy VxDesign, we'll likely use S3 and S3 URLs.